tasks.py

```diff
@@ -552,20 +552,33 @@ def lint(ctx, env="container"):
 
 
 @task(help={
-    "name": "name of the kind cluster to test.",
-    "export": "where to export kind logs"
+    "name": "name of the kind cluster to test (only kind uses).",
+    "export": "where to export kind logs.",
+    "kubeconfig_path": "kubeconfig path. By default, load kubeconfig from kind.",
+    "system_namespace": "comma separated kubernetes system namespace list.",
+    "service_pod_port": "port number that service pods open.",
+    "skip_docker": "don't use docker command in BGP testing."
 })
-def e2etest(ctx, name="kind", export=None):
+def e2etest(ctx, name="kind", export=None, kubeconfig_path=None, system_namespace="kube-system,metallb-system", service_pod_port=80, skip_docker=False):
     """Run E2E tests against development cluster."""
-    validate_kind_version()
-    clusters = run("kind get clusters", hide=True).stdout.strip().splitlines()
-    if name in clusters:
-        kubeconfig = tempfile.NamedTemporaryFile()
-        run("kind export kubeconfig --name={} --kubeconfig={}".format(name, kubeconfig.name), pty=True, echo=True)
+    if skip_docker:
+        opt_skip_docker = "-skip-docker"
     else:
-        raise Exit(message="Unable to find cluster named: {}".format(name))
+        opt_skip_docker = ""
+
+    if kubeconfig_path is None:
+        validate_kind_version()
+        clusters = run("kind get clusters", hide=True).stdout.strip().splitlines()
+        if name in clusters:
+            kubeconfig = tempfile.NamedTemporaryFile()
+            kubeconfig_path = kubeconfig.name
+            run("kind export kubeconfig --name={} --kubeconfig={}".format(name, kubeconfig.name), pty=True, echo=True)
+        else:
+            raise Exit(message="Unable to find cluster named: {}".format(name))
+    else:
+        os.environ['KUBECONFIG'] = kubeconfig_path
```

Comment:
It seems like you use the env variable to define the kubeconfig location here but the kubeconfig file above. Can you make it consistent between both paths?

Reply:
above kubeconfig , ` Run("kind export kubeconfig --name={} --kubeconfig={}".format(name, kubeconfig.name), pty=True, echo=True)` is to export (i.e. generate) kubeconfig from kind cluster, I suppose. `os.environ['KUBECONFIG'] = kubeconfig_path` is to set the specific kubeconfig files for following `kubectl` command. 

In Kind cluster, kubeconfig file is temporally file generated by `tempfile.NamedTemporaryFile()`, however, non kind cluster uses usual file (not temporally file), so we cannot make it consistent because kind's kubeconfig and non-kind's kubeconfig are totally different.

Reply:
My main issue (nit) was in ```go test --provider=local -ginkgo.focus=BGP --kubeconfig={} -service-pod-port={} {}".format(kubeconfig, service_pod_port, opt_skip_docker))``` we pass the kubeconfig as a parameter and I thought we could just use the environment variable and remove that flag. However, I just tried and it doesn't work. I think what you have here is fine.